### PR TITLE
Add mps implementation for unique(no dim support) and unique_consecutive

### PR DESCRIPTION
**Pull request by chrisbbayley**
Adds MPSGraph implementation for unique and unique_consecutive. Currently not possible to support unique with dim argument, as these requires sorting with tensor keys, rather than scalar keys.

### CODE REVIEW

**Review by DenisVieriu97 — COMMENTED** [review]
Could you please change the target branch to master? We have recently switched back to `master` (mps_master is deprecated going forward)

File: aten/src/ATen/native/mps/operations/Unique.mm [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    for(NSUInteger i = 0; i < [shape count]; i++)
+      length *= [shape[i] integerValue];
+    destShape = @[[NSNumber numberWithUnsignedInteger:length]];
+  }
+
+  MPSGraphTensor *sortedInput = nil;
+  if (consecutive)
+    sortedInput = inputTensor;
+  else
+    sortedInput = [graph sortWithTensor:inputTensor
```
[the remark this answers is not in the document]
Comment (kulinseth) [reply aten/src/ATen/native/mps/operations/Unique.mm:84]: @DenisVieriu97 , can you please help with this.?

[the remark this answers is not in the document]
Comment (DenisVieriu97) [reply aten/src/ATen/native/mps/operations/Unique.mm:84]: Sure 👍

**Review by kulinseth — APPROVED** [review]
Modulo the changes to make sure the Ventura side changes are made behind OS flags. it looks good.

**Comment by kulinseth** [comment]
@DenisVieriu97 , also please add TestConsistency tests for unique.

**Review by DenisVieriu97 — APPROVED** [review]

**Pull request merged**